src/herder/TransactionQueue.cpp

```diff
@@ -459,9 +460,15 @@ TransactionQueue::canAdd(
             mApp.getLedgerManager().getLastClosedLedgerNum() + 1;
     }
 
-    auto txResult =
-        tx->checkValid(mApp.getAppConnector(), ls, 0, 0,
-                       getUpperBoundCloseTimeOffset(mApp, closeTime));
+    // Loadgen txs were generated by this local node, and therefore can skip
+    // validation, and be added directly to the queue.
+    auto txResult = tx->createSuccessResult();
```

Comment:
Can we refactor this logic behind an ifdef? It seems fairly dangerous to have logic to skip verification available in non-test builds. I think we can pipe the flag everywhere without the ifdef, but only use it inside the ifdef.

Reply:
Agreed, done